File: app/src/main/java/com/futabooo/android/booklife/screen/BookListMenu.java
package com.futabooo.android.booklife.screen;

import com.futabooo.android.booklife.R;

public enum BookListMenu {
  READ(0, R.string.book_read,"read"),
  READING(1, R.string.book_reading,"reading"),
  TO_READ(2, R.string.book_to_read,"wish"),
  QUITTED(3, R.string.book_quitted,"stacked");

  private final int position;
  private final int titleResId;
  private final String key;

  BookListMenu(int position, int titleResId, String key){
    this.position = position;
    this.titleResId = titleResId;
    this.key = key;
  }

  public static BookListMenu fromPosition(int position) {
    for (BookListMenu bookListMenu : values()) {
      if (position == bookListMenu.getPosition()) {
        return bookListMenu;
      }
    }
    return null;
  }

  public int getPosition() {
    return position;
  }

  public int getTitleResId() {
    return titleResId;
  }

  public String getKey() {
    return key;
  }
}

File: app/src/main/kotlin/com/futabooo/android/booklife/screen/BookListMenu.kt
package com.futabooo.android.booklife.screen

import com.futabooo.android.booklife.R

enum class BookListMenu constructor(val position: Int, val titleResId: Int, val key: String) {
  READ(0, R.string.book_read, "read"),
  READING(1, R.string.book_reading, "reading"),
  TO_READ(2, R.string.book_to_read, "wish"),
  QUITTED(3, R.string.book_quitted, "stacked");

  companion object {
    fun fromPosition(position: Int): BookListMenu = values().first { position == it.position }
  }
}
